receiver/awscontainerinsightreceiver/internal/host/nodeCapacity.go

```diff
@@ -52,26 +49,25 @@ func newNodeCapacity(logger *zap.Logger, options ...nodeCapacityOption) (nodeCap
 	if _, err := nc.osLstat(hostProc); os.IsNotExist(err) {
 		return nil, err
 	}
-	if err := nc.osSetenv(goPSUtilProcDirEnv, hostProc); err != nil {
-		return nil, fmt.Errorf("NodeCapacity cannot set goPSUtilProcDirEnv to %s: %w", hostProc, err)
-	}
+	envMap := common.EnvMap{common.HostProcEnvKey: hostProc}
+	ctx := context.WithValue(context.Background(), common.EnvKey, envMap)
```

Comment:
Does this need to extract the current value and merge it with the new value?

Reply:
What current value? Sorry I don't quite follow.

Reply:
Sorry, I wasn't paying close enough attention.  I expected this to be operating on an existing context, not conjuring one from the air.  If it were operating on an existing context then it would be necessary to get the value in that context at `common.EnvKey` and modify it before setting that value into the context to be used in order to avoid inadvertently changing other values in that object that are necessary for the correct functioning of the code utilizing it.

Reply:
that makes sense, and maybe something you could do eventually, but you're better off this way right now with an explicit config key for hostProc hopefully!